dbms/src/Flash/FlashService.cpp

```diff
@@ -137,7 +137,31 @@ grpc::Status FlashService::Coprocessor(
 
     context->setMockStorage(mock_storage);
 
+    auto settings = context->getSettingsRef();
+    auto handle_limit = settings.cop_pool_handle_limit != 0 ? settings.cop_pool_handle_limit.get() : 10 * cop_pool->size();
+    auto max_queued_duration_seconds = settings.cop_pool_max_queued_seconds;
+
+    // We use this atomic variable metrics from the prometheus-cpp library to mark the number of queued queries.
+    // TODO: Use grpc asynchronous server and a more full-featured thread pool.
+    if (handle_limit > 0)
+    {
+        if (auto current = GET_METRIC(tiflash_coprocessor_handling_request_count, type_cop).Value(); current > handle_limit)
+        {
+            response->mutable_region_error()->mutable_server_is_busy()->set_reason(fmt::format("tiflash cop pool queued too much, current = {}, limit = {}", current, handle_limit));
+            return grpc::Status::OK;
+        }
+    }
+
+
     grpc::Status ret = executeInThreadPool(*cop_pool, [&] {
+        if (max_queued_duration_seconds > 0)
```

Comment:
should it also applied to BatchCoprocessor?

Reply:
I think we needn't, because:
1. most pressure is from remote read, which is cop request.
2. batch cop will be deprecated soon.